network-static-2-httpks.ks.in

```diff
@@ -0,0 +1,68 @@
+#test name: network-static-2
+#various combinations of activation in initramfs and stage 2
+url @KSTEST_URL@
+install
+# activated in initramfs with dhcp, reactivated in stage 2
+network --device=ens4 --bootproto static --ip @KSTEST_STATIC_IP1@ --netmask @KSTEST_STATIC_NETMASK@ --gateway @KSTEST_STATIC_GATEWAY@
+# activated in initramfs with dhcp, not reactivated in stage 2
+network --device=ens5 --bootproto static --ip @KSTEST_STATIC_IP2@ --netmask @KSTEST_STATIC_NETMASK@ --gateway @KSTEST_STATIC_GATEWAY@
+# not activated in initramfs with dhcp, reactivated in stage 2
+network --device=ens6 --bootproto static --ip @KSTEST_STATIC_IP3@ --netmask @KSTEST_STATIC_NETMASK@ --gateway @KSTEST_STATIC_GATEWAY@ --activate
+# not activated in initramfs with dhcp, not reactivated in stage 2
+network --device=ens7 --bootproto static --ip @KSTEST_STATIC_IP4@ --netmask @KSTEST_STATIC_NETMASK@ --gateway @KSTEST_STATIC_GATEWAY@
+
+bootloader --timeout=1
+zerombr
+clearpart --all
+autopart
+
+keyboard us
+lang en
+timezone America/New_York
+rootpw qweqwe
+shutdown
+
+%packages
+%end
+
+%post --nochroot
+
+@KSINCLUDE@ post-nochroot-lib-network.sh
+
+check_gui_configurations ens3 ens4 ens5 ens6 ens7
+
+%end
+
+%post
+
+@KSINCLUDE@ post-lib-network.sh
+
+check_device_connected ens3 yes
+
+check_device_ifcfg_value ens4 IPADDR @KSTEST_STATIC_IP1@
+check_device_connected ens4 yes
+check_device_ipv4_address ens4 @KSTEST_STATIC_IP1@
+
+check_device_ifcfg_value ens5 IPADDR @KSTEST_STATIC_IP2@
+check_device_connected ens5 yes
+
+check_device_ifcfg_value ens6 IPADDR @KSTEST_STATIC_IP3@
+check_device_connected ens6 yes
+check_device_ipv4_address ens6 @KSTEST_STATIC_IP3@
+
+check_device_ifcfg_value ens7 IPADDR @KSTEST_STATIC_IP4@
+if [[ "@KSTEST_OS_NAME@" == "rhel" ]]; then
+    check_device_connected ens7 no
+else
+    # Fedora autoactivates default connections (Wired Connection 2)
+    # if it doesn't find ifcfg for the device, which is the case for
+    # kickstart in %pre (unlike normal kickstart where the ifcfg file
+    # is created in initramfs.
```

Comment:
Oh, and I have also to update the content of network-static-2-pre.sh (move the code from network-static-2.sh instead of including it) because I am not adding the network-static-2.sh test to our suite.

The network-static-2.sh would behave strange (because ks is injected in initramfs) similar to some other tests (this is why we have the httpks versions of the tests) as described in the issue https://projects.engineering.redhat.com/browse/INSTALLER-272. So I will keep just network-static.sh for the special case of ks injected in initramfs and for network-static-2 I will add just -httpks and -pre versions of the test.